native/src/tablecloth.mli

```diff
@@ -728,34 +728,855 @@ module Char : sig
+
+  val infinity : t
+  (** Positive {{: https://en.wikipedia.org/wiki/IEEE_754-1985#Positive_and_negative_infinity } infinity } 
+
+    {[Float.log ~base:10.0 0.0 = Float.infinity]} *)
+  
+  val negativeInfinity : t
+  (** Negative infinity, see {!Float.infinity} *)
+
+  val negative_infinity : t
+
+  val e : t
+  (** An approximation of {{: https://en.wikipedia.org/wiki/E_(mathematical_constant) } Euler's number }. *)
+
+  val pi : t
+  (** An approximation of {{: https://en.wikipedia.org/wiki/Pi } pi }. *)
+
+  (** {1 Basic arithmetic and operators} *)
+
+  val add : t -> t -> t
+  (** Addition for floating point numbers.
+
+    {[Float.add 3.14 3.14 = 6.28]}
+
+    {[Float.(3.14 + 3.14 = 6.28)]}
+
+    Although [int]s and [float]s support many of the same basic operations such as 
+    addition and subtraction you {b cannot} [add] an [int] and a [float] directly which 
+    means you need to use functions like {!Int.toFloat} or {!Float.roundToInt} to convert both values to the same type.
+
+    So if you needed to add a {!List.length} to a [float] for some reason, you
+    could:
+
+    {[Float.add 3.14 (Int.toFloat (List.length [1,2,3])) = 6.14]}
+
+    or
+
+    {[Float.roundToInt 3.14 + List.length [1,2,3] = 6]}
+
+    Languages like Java and JavaScript automatically convert [int] values
+    to [float] values when you mix and match. This can make it difficult to be sure
+    exactly what type of number you are dealing with and cause unexpected behavior.
+
+    OCaml has opted for a design that makes all conversions explicit.
+  *)
+
+  val ( + ) : t -> t -> t
+  (** See {!Float.add} *)
+
+  val subtract : t -> t -> t
+  (** Subtract numbers 
+    {[Float.subtract 4.0 3.0 = 1.0]}
+
+    Alternatively the operator can be used:
+
+    {[Float.(4.0 - 3.0) = 1.0]}    
+  *)
+
+  val ( - ) : t -> t -> t
+  (** See {!Float.subtract} *)
+
+  val multiply : t -> t -> t
+  (** Multiply numbers like 
+
+    {[Float.multiply 2.0 7.0 = 14.0]} 
+
+    Alternatively the operator can be used:
+
+    {[Float.(2.0 * 7.0) = 14.0]}     
+  *)
+
+  val ( * ) : t -> t -> t
+  (** See {!Float.multiply} *)
+
+  val divide : t -> t -> t
+  (** Floating-point division:
+
+    {[Float.divide 3.14 2.0 = 1.57]} 
+
+    Alternatively the operator can be used:
+
+    {[Float.(3.14 / 2.0) = 1.57]} *)
+
+  val ( / ) : t -> t -> t
+  (** See {!Float.divide} *)
+
+  val power : t -> t -> t
+  (** Exponentiation, takes the base first, then the exponent. 
+
+    {[Float.power 7.0 3.0 = 343.0]}
+
+    Alternatively the operator can be used:
+
+    {[Float.(7.0 ** 3.0) = 343.0]}
+  *)
+
+  val ( ** ) : t -> t -> t
+  (** See {!Float.power} *)
+
+  val negate : t -> t
+  (** Flips the 'sign' of a [float] so that positive floats become negative and negative integers become positive. Zero stays as it is.
+
+    {[Float.negate 8 = (-8)]}
+
+    {[Float.negate (-7) = 7]}
+
+    {[Float.negate 0 = 0]} 
```

Comment:
Sure about this? Isn't the -0 in floats? Can we test this?

Reply:
That’s correct, but -0.0 = 0.0 evaluates to true.

I intend to turn all of the examples into tests.